### ISSUE TITLE
`get_log_probs` returns shifted log probabilities while every caller masks them with unshifted labels

### ISSUE THREAD

**Issue by CodersAcademy006** [ellanorai/ThinkRL#84]
This one needs a decision before any code change, because fixing it changes every number the project has produced.

`thinkrl/algorithms/base.py:291-309` performs the causal shift and then appends a zero column:

```python
shift_logits = logits[:, :-1, :]          # logits[t] predicts token t+1
shift_labels = labels[:, 1:]
token_log_probs = gather(log_softmax(shift_logits), shift_labels)
token_log_probs = token_log_probs * (shift_labels != -100).float()
padding = torch.zeros(B, 1)
return torch.cat([token_log_probs, padding], dim=1)     # [B, S], left-aligned
```

Index `t` of the result holds `log p(labels[t+1])`. The docstring describes the return as "Log probabilities [B, S] with 0.0 at masked positions", which implies alignment with `labels`. Callers rely on that reading and mask with the unshifted labels, for example `reinforce_pp.py:237`:

```python
token_mask = (labels != -100).float()     # unshifted
```

used against the shifted log probabilities at lines 288, 306 and 346. `grep -rn "labels != -100" thinkrl/algorithms/*.py` gives 14 occurrences across 11 files: base, grpo, dr_grpo, dapo, papo, reinforce_pp, star, ipo, reinforce, vapo and ppo.

Worked example on a six-token sequence with a two-token prompt:

```
labels          : [-100, -100, 2, 3, 4, 1]
get_log_probs   : [-0.0, -6.106, -0.402, -0.932, -1.317, 0.0]
mask (unshifted): [0, 0, 1, 1, 1, 1]
lp * mask       : [-0.0, -0.0, -0.402, -0.932, -1.317, 0.0]
```

The log probability of the first generated token, `-6.106`, sits at index 1 and the mask does not select it. Index 5 is the appended zero and the mask does select it. The masked mean is `-0.663`; the correct mean over the four generated tokens is `-2.189`. The dropped token tends to carry the largest magnitude because it is the least predictable position in the completion.

This is not confined to existing code. The `feat/kto-clean` branch, the newest work in the repository, reproduces the same pattern at `kto.py:140`, `:146` and `:149`. That suggests the problem is the contract rather than eleven independent mistakes, and that each new algorithm will inherit it.

**What I would like to know before proposing anything**

Would you prefer to correct the alignment and re-run the benchmarks, correct it behind a flag so existing numbers remain reproducible, or leave it and document the current behaviour as intended? I have not touched this and will not open a PR until you have decided, since it moves every published result including REINFORCE++.

@Archit03
